formats/Filtered/libs/ext.srf.filtered.value-filter.js

```diff
@@ -254,11 +262,14 @@
 				filtercontrols.height( height );
 			}
 
-
 			// insert options (checkboxes and labels) and attach event handlers
 			// TODO: Do we need to wrap these in a form?
-			for ( var j in sortedDistinctValues ) {
+			for (var j = 0; j < sortedDistinctValues.length; j++) {
 				var option = $('<div class="filtered-value-option">');
+
+				// Try to get label, if not fall back to value id
+				var label = distinctValues[sortedDistinctValues[j]] || sortedDistinctValues[j];
+
 				var checkbox = $('<input type="checkbox" class="filtered-value-value" value="' + sortedDistinctValues[j] + '"  >');
```

Comment:
And I agree about the code quality. The whole file is just one function.

But procedural code isn't always that bad, at least its easy to understand and to reason about. You can also overengineer things to the point that they're even more of a maintainence burden than good old sphaghetti code ;) But enough philosophy here.